dsl_parser/blueprint_diff.py

```diff
@@ -0,0 +1,120 @@
+########
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+
+def blueprint_diff(old_blueprint, new_blueprint):
+
+    path_of_keys = []
+    output = {}
+
+    diff(old_blueprint, new_blueprint, path_of_keys, output, 'added')
+    diff(new_blueprint, old_blueprint, path_of_keys, output, 'removed')
+
+    return output
+
+
+def diff(old, new, path_of_keys, output, mode):
```

Comment:
if mode is binary, it's best to just be a boolean argument with a semantic change